src/OpenTelemetry/OpenTelemetrySdkExtensions.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Contains methods for extending the <see cref="OpenTelemetrySdk"/> class.
+/// </summary>
+public static class OpenTelemetrySdkExtensions
+{
+    private static readonly NullLoggerFactory NoopLoggerFactory = new();
+
+    /// <summary>
+    /// Gets the <see cref="ILoggerFactory"/> contained in an <see
+    /// cref="OpenTelemetrySdk"/> instance.
+    /// </summary>
+    /// <remarks>
+    /// Note: The default <see cref="ILoggerFactory"/> will be a no-op instance.
+    /// Call <see
+    /// cref="OpenTelemetryBuilderSdkExtensions.WithLogging(IOpenTelemetryBuilder)"/>
+    /// to enable logging.
+    /// </remarks>
+    /// <param name="sdk"><see cref="OpenTelemetrySdk"/>.</param>
+    /// <returns><see cref="ILoggerFactory"/>.</returns>
+    public static ILoggerFactory GetLoggerFactory(this OpenTelemetrySdk sdk)
```

Comment:
Exploring the .NET Framework and console app use cases for a sec...

If a user has already adopted ILogger as their logging framework, then they probably have code somewhere like

```csharp
var loggerFactory = LoggerFactory.Create((loggingBuilder) => {
    loggingBuilder.AddConsole() ...
});
``` 

If they then use `WithLogging`, they'll need to refactor things to avoid having to interact with two LoggerFactory instances to get loggers.

Are there other options that might minimize the need for refactoring? Would an additional overload of `WithLogging` accepting a LoggerFactory as an argument help? My assumption is that if they pass in their existing LoggerFactory then we'd be able to add the OpenTelemetry logger provider.

Reply:
> Would an additional overload of WithLogging accepting a LoggerFactory as an argument help?

I hesitate to do this. There is an [ILoggerFactory.AddProvider](https://learn.microsoft.com/dotnet/api/microsoft.extensions.logging.iloggerfactory.addprovider) API. BUT it is after `IServiceProvider` has been created. We won't be able to add anything to the `IServiceCollection` and we'll have to bootstrap manually everything `OpenTelemetryLoggerProvider` and `LoggerProvider` need as well as _anything_ _any_ component might need (ex `OtlpLogExporter`). Could be possible, but it is going to be a big headache and create a sort of parallel universe we'll have to deal with for eternity.

There is also an issue if we add an extension something like `WithLogging(this IOpenTelemetryBuilder builder, ILoggerFactory loggerFactory)` that will also appear to hosting users where we probably don't want it.

What other options do we have?

* We could expose the currently experimental `ILoggingBuilder.UseOpenTelemetry` extensions

  This would allow people to continue to use `LoggerFactory.Create` style and utilize `LoggerProviderBuilder`. Which is to say, we don't force anyone to switch to `WithLogging`.

* Give different migration guidance

   Let's say .NET Framework user has this today:
   
   ```csharp
   using var loggerFactory = LoggerFactory.Create((loggingBuilder) => {
       loggingBuilder.AddConsole().AddOpenTelemetry(...)...
   });
   ```
   
   I guess what you are thinking @alanwest is they would start doing this:
   
   ```csharp
   using var loggerFactory = LoggerFactory.Create((loggingBuilder) => {
       loggingBuilder.AddConsole()...
   });
   
   using var openTelemetry = OpenTelemetrySdk.Create(builder =>
       builder.WithLogging(...));
   
   var otelLoggerFactory = openTelemetry.GetLoggerFactory();
   ```
   
   In that case they would have two logger factories and need to do some work.
   
   But they could do this instead:
   
   ```csharp
   using var openTelemetry = OpenTelemetrySdk.Create(builder =>
   {
       builder.Services.AddLogging(logging => logging.AddConsole()...);
   
       builder.WithLogging(...);
   });
   
   var loggerFactory = openTelemetry.GetLoggerFactory();
   ```
   
   They'll end up with a single factory with Console + OTel as they had before.
   
   We could simplify that a bit with another `WithLogging` overload like...
   
   ```csharp
   public static IOpenTelemetryBuilder WithLogging(
      this IOpenTelemetryBuilder builder,
      Action<LoggerProviderBuilder>? configureBuilder,
      Action<OpenTelemetryLoggerOptions>? configureOptions,
      Action<ILoggingBuilder>? configureILogger)
   { }
   ```
   
   Then they could do:
   
   ```csharp
   using var openTelemetry = OpenTelemetrySdk.Create(builder =>
   {
       builder.WithLogging(
           logging => logging.AddOtlpExporter(),
           options => options.IncludeFormattedMessage = true,
           ilogger => ilogger.AddConsole());
   });
   
   var loggerFactory = openTelemetry.GetLoggerFactory();
   ```
   
   It is a kind of heavy/complicated API though. Do we think this is common enough/there will be enough demand to justify a dedicated API?

* We could just wait and see what the feedback is and if there is demand for something.